src/Nox.ClientApi.Tests/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Queries/GetWorkplacesQuery.g.cs

```diff
@@ -8,33 +8,64 @@
 using Nox.Application.Commands;
 
 using ClientApi.Application.Dto;
-using ClientApi.Infrastructure.Persistence;
+using ClientApi.Infrastructure.Persistence;using Nox.Presentation.Api;
+using Nox.Solution;
 
 namespace ClientApi.Application.Queries;
 
 public record GetWorkplacesQuery() : IRequest<IQueryable<WorkplaceDto>>;
 
 internal partial class GetWorkplacesQueryHandler: GetWorkplacesQueryHandlerBase
 {
-    public GetWorkplacesQueryHandler(DtoDbContext dataDbContext): base(dataDbContext)
+    public GetWorkplacesQueryHandler(DtoDbContext dataDbContext,
+        NoxSolution solution,
+        IHttpLanguageProvider languageProvider): base(dataDbContext,
+            solution,
+            languageProvider)
     {
     
     }
 }
 
 internal abstract class GetWorkplacesQueryHandlerBase : QueryBase<IQueryable<WorkplaceDto>>, IRequestHandler<GetWorkplacesQuery, IQueryable<WorkplaceDto>>
-{
-    public  GetWorkplacesQueryHandlerBase(DtoDbContext dataDbContext)
+{private readonly NoxSolution _solution;
+        private readonly IHttpLanguageProvider _languageProvider;
+
+    public  GetWorkplacesQueryHandlerBase(DtoDbContext dataDbContext,
+        NoxSolution solution,
+        IHttpLanguageProvider languageProvider)
     {
-        DataDbContext = dataDbContext;
+        DataDbContext = dataDbContext;_solution = solution;
+        _languageProvider = languageProvider;
     }
 
     public DtoDbContext DataDbContext { get; }
 
     public virtual Task<IQueryable<WorkplaceDto>> Handle(GetWorkplacesQuery request, CancellationToken cancellationToken)
     {
-        var item = (IQueryable<WorkplaceDto>)DataDbContext.Workplaces
-            .AsNoTracking();
-       return Task.FromResult(OnResponse(item));
+        var cultureCode = _languageProvider.GetLanguage();
+
+        IQueryable<WorkplaceDto> linqQueryBuilder =
+            from item in DataDbContext.Workplaces.AsNoTracking()
+            join itemLocalizedFromJoin in DataDbContext.WorkplacesLocalized on cultureCode equals itemLocalizedFromJoin.CultureCode into joinedData
+            from itemLocalized in joinedData.Where(l => item.Id == l.Id).DefaultIfEmpty()
```

Comment:
this produces one full join and one left join, one left join should be sufficient, can we discuss this?

Reply:
Done